File: ios/marvel/Data/Models/ApiResponse.swift
//
//  Response.swift
//  marvel
//
//  Created by Vilmar Bispo Filho on 02/06/22.
//

import Foundation

struct ApiResponse<T: Decodable>: Decodable {
    let code: Int
    let status: String
    let copyright: String
    let attributionText: String
    let attributionHTML: String
    let data: DataResponse<T>
    let etag: String
}

File: android_app/data/src/main/java/br/com/data/model/ApiResponse.kt
package br.com.data.model

import com.squareup.moshi.Json

data class ApiResponse<T>(
   @Json(name = "code")
   val code: Int,
   @Json(name = "status")
   val status: String,
   @Json(name = "copyright")
   val copyright: String,
   @Json(name = "attributionText")
   val attributionText: String,
   @Json(name = "attributionHTML")
   val attributionHTML: String,
   @Json(name = "data")
   val data: DataResponse<T>,
   @Json(name = "etag")
   val etag: String
)